libs/network/esp8266/network_esp8266.c

```diff
@@ -1007,14 +1052,42 @@ int net_ESP8266_BOARD_createSocket(
     return SOCKET_ERR_MEM;
   }
 
+  if (socketType & ST_UDP) {
+    pEspconn->type    = ESPCONN_UDP;
+
+    // esp_tcp and esp_udp start identically (up to remote_ip)
+    // so we can leave the proto.tcp as an alias to proto.udp
+  } else {
+    pEspconn->type    = ESPCONN_TCP;
+
+    espconn_set_opt(pEspconn, ESPCONN_NODELAY); // disable nagle, don't need the extra delay
+  }
+
   pSocketData->pEspconn = pEspconn;
-  pEspconn->type      = ESPCONN_TCP;
   pEspconn->state     = ESPCONN_NONE;
   pEspconn->proto.tcp = tcp;
   tcp->remote_port    = port;
   tcp->local_port     = espconn_port(); // using 0 doesn't work
   pEspconn->reverse   = pSocketData;
-  espconn_set_opt(pEspconn, ESPCONN_NODELAY); // disable nagle, don't need the extra delay
+
+  // multicast support
+  // FIXME: perhaps extend the JsNetwork with addmembership/removemembership instead of using options
```

Comment:
Is this something you think would be available on all platforms? I'm struggling to see how I'd implement this for ESP8266... I think maybe we could leave it as-is for now?